// Copyright 2024 The Protoscribe Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// The linguistic grammar for the counting systems.

syntax = "proto2";

package protoscribe;

import "protoscribe/language/morphology/morphology_parameters.proto";

message NumberConfig {
  // Core morphological features (such as gender, number and case information)
  // for the language necessary for generating language's morphemes.
  optional MorphologyParameters core = 1;

  // If enabled then use the version of the number grammar with one-deletion.
  // See `protoscribe.language.morphology.numbers` module.
  optional bool one_deletion = 2 [default = false];

  // The morphological feature configurations below control the verbalization of
  // phrases of the form "number noun".

  // The morphological features for inflecting the type of number: one.
  repeated string one_config_features = 3;

  // The morphological features for inflecting the type of number: two.
  repeated string two_config_features = 4;

  // The morphological features for inflecting the type of number: many.
  repeated string many_config_features = 5;
}
